src/plugins/kibana_utils/public/history/history_observable.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
```

Comment:
Maybe add unit tests for this?

Reply:
It is there 😅 
https://github.com/elastic/kibana/pull/87635/files/6944ac10f785e16a3c11b3e4d0b41ae9135714da#diff-e92c62c77f010c77c6a8dcd294ca0e506d0f586a695bb08848a1e719e9d6e1ec